plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSourceProvider.java

```diff
@@ -560,14 +562,14 @@ public static List<HiveColumnHandle> toColumnHandles(List<ColumnMapping> regular
                                     projectedColumn.getDereferenceIndices(),
                                     projectedColumn.getDereferenceNames(),
                                     fromHiveType,
-                                    fromHiveType.getType(typeManager));
+                                    fromHiveType.getType(typeManager, timestampPrecision));
```

Comment:
Can we get rid of `io.trino.plugin.hive.HiveType#getType(io.trino.spi.type.TypeManager)`, it is not only deprecated but can lead to incorrect results. Follow up maybe?

Reply:
I'm planning to have it as a follow-up